dask/typing.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import annotations
+
+from typing import (
+    Any,
+    Hashable,
+    Mapping,
+    MutableMapping,
+    Protocol,
+    Sequence,
+    TypeVar,
+    runtime_checkable,
+)
+
+try:
+    from IPython.display import DisplayObject
+except ImportError:
+    DisplayObject = object
+
+
+T = TypeVar("T")
+CollectionType = TypeVar("CollectionType", bound="DaskCollection")
+
+
+class PostComputeCallable(Protocol):
+    def __call__(self, partitions: Sequence[Any], *args) -> Any:
+        """ """
+
+
+class PostPersistCallable(Protocol):
+    def __call__(
+        self,
+        dsk: Mapping,
+        *args,
+        rename: Mapping[str, str] | None = None,
+    ) -> DaskCollection:
+        """ """
```

Comment:
I think this could be made generic across `CollectionType`, so 
```suggestion
class PostPersistCallable(Protocol[CollectionType]):
    def __call__(
        self,
        dsk: Mapping,
        *args,
        rename: Mapping[str, str] | None = None,
    ) -> CollectionType:
        """ """
```
This way the `CollectionProtocol` could specify that the post-persist can't just return any dask collection, but one of the same type as the current collection.